### ISSUE TITLE
lesson 7 test_can_pick_winner fails

### ISSUE THREAD

**Issue by mebedel** [smartcontractkit/full-blockchain-solidity-course-py#442]
The video is at 8.16.14

FAILED tests/test_lottery_integration.py::test_can_pick_winner - AssertionError: assert '0x0000000000...00000...
am using kovan testnet 
his is the contract   0x1BBE8Ab9119b1483ec4E08FC94B857873fc3F48c
in the contract event it shows 
OwnershipTransferred (index_topic_1 address previousOwner, index_topic_2 address newOwner)[topic0] 0x8be0079c531659141344cd1fd0a4f28419497f9722a3daafe3b4186f6b6457e0[topic1] 0x0000000000000000000000000000000000000000000000000000000000000000[topic2] 0x0000000000000000000000009be24bee912a424cc60c1c37cb0eb33196fbe4a4

my account doesn't get the money as supposed instead money goest to this account 
0x0000000000000000000000009be24bee912a424cc60c1c37cb0eb33196fbe4a4
which doesn't exist 

my lottery.sol is different from the video in few spots:
         // original pragma sol  is 0.6.6
        1. pragma solidity >=0.6.6 <0.9.0; 
        // i changed v0.6 to v0.8
        2. import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
        // nothing got changed
        3. import "@openzeppelin/contracts/access/Ownable.sol";
        // i changed v0.6 to v0.8
        4. import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol"; 
        // i changed players.push(msg.sender); to what is below
        5. players.push(payable(msg.sender));
      
I am essentially unable to pay the winner
whats should I do?
from: long time lurker first time Questioner

**Comment by cromewar** [comment]
This happens due to the time it takes to pick the winner, you are getting the winner as 0x000... which is the immediate ressponse you got, but the transaction actually take longer, you might solve this adding a tx.wait(300) and wait more to get the correct winner address, or use something like web3 async.

**Comment by mebedel** [comment]
I’ll try it thanks

**Comment by CincyGhost** [comment]
Hello, I'm having exact same problem.....did this work for you?   If it did work can you post a copy of your code? thnx

**Comment by cromewar** [comment]
@CincyGhost Give it a try, you can clone the code from the repo ans use tx.wait(300).
Best regards, Cromewar.

**Comment by noopurphalak** [comment]
@cromewar Where exactly should `tx.wait(300)` be placed?

**Comment by cromewar** [comment]
@noopurphalak on your python script, if you make a request to for contract, like deploying it or executing a function, you can add that process to a variable (tx for ocnvention) and then ask it to wait until is finished with tx.wait(time).

**Comment by noopurphalak** [comment]
@cromewar I am aware of that, What my question was should put `tx.wait(300)` for every transaction in the code or only to a specific one to solve this issue?

**Comment by cromewar** [comment]
No, just for specific ones. like selecting the winner of a lottery, or something you know will take time, the transaction will be executed anyways on the blockchain, but you wont want the response on your print statement. A more elegant solution would be to use web3 async to get synchronous responses.
